topostats/processing.py

```diff
@@ -604,99 +608,69 @@ def run_nodestats(
-        LOGGER.info(f"[{filename}] : *** Nodestats ***")
-        nodestats_image_data = defaultdict()
+    if ordered_tracing_config["run"]:
+        ordered_tracing_config.pop("run")
+        LOGGER.info(f"[{filename}] : *** Ordered Tracing ***")
+        ordered_tracing_image_data = defaultdict()
         grainstats_additions_image = pd.DataFrame()
-        try:
-            # run image using directional grain masks
-            for direction, disordered_tracing_direction_data in disordered_tracing_data.items():
-                (
-                    nodestats_data,
-                    grainstats_additions_df,
-                    nodestats_full_images,
-                    nodestats_branch_images,
-                ) = nodestats_image(
-                    image=image,
-                    disordered_tracing_direction_data=disordered_tracing_direction_data,
-                    filename=filename,
-                    pixel_to_nm_scaling=pixel_to_nm_scaling,
-                    **nodestats_config,
-                )
 
-                # save per image new grainstats stats
-                grainstats_additions_df["threshold"] = direction
-                grainstats_additions_image = pd.concat([grainstats_additions_image, grainstats_additions_df])
+        # try:
+        # run image using directional grain masks
+        for direction, disordered_tracing_direction_data in disordered_tracing_data.items():
+            (
+                ordered_tracing_data,
+                grainstats_additions_df,
+                ordered_tracing_full_images,
+            ) = ordered_tracing_image(
+                image=image,
+                disordered_tracing_direction_data=disordered_tracing_direction_data,
+                nodestats_direction_data=nodestats_data[direction],
+                filename=filename,
+                pixel_to_nm_scaling=pixel_to_nm_scaling,
+                **ordered_tracing_config,
+            )
 
-                # append direction results to dict
-                nodestats_image_data[direction] = nodestats_data
+            # save per image new grainstats stats
+            grainstats_additions_df["threshold"] = direction
+            grainstats_additions_image = pd.concat([grainstats_additions_image, grainstats_additions_df])
 
-                # save whole image plots
+            # append direction results to dict
+            ordered_tracing_image_data[direction] = ordered_tracing_data
+
+            # save whole image plots
+            Images(
+                filename=f"{filename}_{direction}_ordered_traces",
+                data=image,
+                masked_array=ordered_tracing_full_images.pop("ordered_traces"),
+                output_dir=core_out_path,
+                **plotting_config["plot_dict"]["ordered_traces"],
+            ).plot_and_save()
+            # save optional false core_set plots
+            for plot_name, image_value in ordered_tracing_full_images.items():
```

Comment:
It should be in the comment above as the loop aims to plot images with the `core_set` set to `false`. So this would be specified in the plotting config if you want to plot the core or all set images.

I've amended this to be clearer saying: "save optional diagnostic plots (those with core_set = False)"